From 9b61a81f0f4a93e9e1615c50a66abc8537ea2adb Mon Sep 17 00:00:00 2001
From: Brandon Hong <brandon.hong@intel.com>
Date: Wed, 2 Oct 2019 10:53:50 +0800
Subject: [PATCH 1/3] Add sysmacros header

Add sysmacros header to fix build error for undefined major(), minor() or makedev().

Signed-off-by: Brandon Hong <brandon.hong@intel.com>
---
 grub-core/osdep/devmapper/getroot.c  | 2 ++
 grub-core/osdep/devmapper/hostdisk.c | 1 +
 grub-core/osdep/linux/getroot.c      | 2 ++
 grub-core/osdep/unix/getroot.c       | 2 ++
 4 files changed, 7 insertions(+)

diff --git a/grub-core/osdep/devmapper/getroot.c b/grub-core/osdep/devmapper/getroot.c
index d02386b..a726c2a 100644
--- a/grub-core/osdep/devmapper/getroot.c
+++ b/grub-core/osdep/devmapper/getroot.c
@@ -50,6 +50,8 @@
 #include <grub/emu/misc.h>
 #include <grub/emu/hostdisk.h>
 
+#include <sys/sysmacros.h>
+
 static int
 grub_util_open_dm (const char *os_dev, struct dm_tree **tree,
 		   struct dm_tree_node **node)
diff --git a/grub-core/osdep/devmapper/hostdisk.c b/grub-core/osdep/devmapper/hostdisk.c
index 19c1101..4e3aec3 100644
--- a/grub-core/osdep/devmapper/hostdisk.c
+++ b/grub-core/osdep/devmapper/hostdisk.c
@@ -24,6 +24,7 @@
 #include <errno.h>
 #include <limits.h>
 
+#include <sys/sysmacros.h>
 
 #ifdef HAVE_DEVICE_MAPPER
 # include <libdevmapper.h>
diff --git a/grub-core/osdep/linux/getroot.c b/grub-core/osdep/linux/getroot.c
index 772de0a..cf7a3ad 100644
--- a/grub-core/osdep/linux/getroot.c
+++ b/grub-core/osdep/linux/getroot.c
@@ -57,6 +57,8 @@
 #include <grub/emu/exec.h>
 #include <grub/btrfs.h>
 
+#include <sys/sysmacros.h>
+
 #define LVM_DEV_MAPPER_STRING "/dev/mapper/"
 
 /* Defines taken from btrfs/ioctl.h.  */
diff --git a/grub-core/osdep/unix/getroot.c b/grub-core/osdep/unix/getroot.c
index e3887cb..9710dc6 100644
--- a/grub-core/osdep/unix/getroot.c
+++ b/grub-core/osdep/unix/getroot.c
@@ -69,6 +69,8 @@
 #include <grub/emu/hostdisk.h>
 #include <grub/emu/getroot.h>
 
+#include <sys/sysmacros.h>
+
 #if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 # define MAJOR(dev) major(dev)
 # define FLOPPY_MAJOR	2
-- 
2.23.0

